File: terminal_client/lib/ui/idle_main_layer_placeholder.dart
import 'package:terminal_client/gen/terminals/ui/v1/ui.pb.dart' as uiv1;

/// Canonical main-layer placeholder until the first `SetUI` (terminal-ui
/// plan, Phase H). Wire shape must match [ui.IdleMainLayerPlaceholder] on the
/// server; `TestIdleMainLayerPlaceholderGoldenWire` guards the Go side and
/// `idle_main_layer_placeholder_test.dart` asserts semantic parity with the
/// unmarshaled golden protobuf.
uiv1.Node idleMainLayerPlaceholderRoot() {
  return uiv1.Node()
    ..id = '__runtime.main_placeholder.root'
    ..props['client_chrome'] = 'hidden'
    ..props['background'] = '#101418'
    ..props['type'] = 'stack'
    ..stack = uiv1.StackWidget()
    ..children.add(
      uiv1.Node()
        ..id = '__runtime.main_placeholder.message'
        ..props['type'] = 'text'
        ..props['value'] = 'Awaiting server UI'
        ..props['style'] = 'headline'
        ..props['color'] = '#E7F0F7'
        ..text = (uiv1.TextWidget()
          ..value = 'Awaiting server UI'
          ..style = 'headline'
          ..color = '#E7F0F7'),
    );
}

File: terminal_server/internal/ui/idle_main_layer_placeholder.go
package ui

// IdleMainLayerPlaceholder returns the canonical main-layer UI tree shown by
// thin clients after control registration until the first server SetUI
// arrives (plans/features/terminal-ui/plan.md, Phase H). The shape is mirrored
// on the Flutter client for display-surface mode; keep wire bytes in sync
// with transport/testdata/idle_main_layer_placeholder_root.pb (see
// TestIdleMainLayerPlaceholderGoldenWire in internal/transport).
func IdleMainLayerPlaceholder() Descriptor {
	return Descriptor{
		Type: "stack",
		ID:   "__runtime.main_placeholder.root",
		Props: map[string]string{
			"client_chrome": "hidden",
			"background":    "#101418",
		},
		Children: []Descriptor{
			{
				Type: "text",
				ID:   "__runtime.main_placeholder.message",
				Props: map[string]string{
					"value": "Awaiting server UI",
					"style": "headline",
					"color": "#E7F0F7",
				},
			},
		},
	}
}
